framework/src/modules/bft/api.ts

```diff
@@ -12,45 +12,253 @@
+		height: number,
+	): Promise<number> {
+		const paramsStore = context.getStore(this.moduleID, STORE_PREFIX_BFT_PARAMETERS);
+		const start = intToBuffer(height + 1, 4, BIG_ENDIAN);
+		const end = intToBuffer(MAX_UINT32, 4, BIG_ENDIAN);
+		const results = await paramsStore.iterate({
+			limit: 1,
+			start,
+			end,
+		});
+		if (results.length !== 1) {
+			throw new BFTParameterNotFoundError();
+		}
+		const [result] = results;
+		return result.key.readUInt32BE(0);
+	}
+
+	public async setBFTParameters(
+		context: APIContext,
+		precommitThreshold: bigint,
+		certificateThreshold: bigint,
+		validators: Validator[],
+	): Promise<void> {
+		let aggregateBFTWeight = BigInt(0);
+		for (const validator of validators) {
+			if (validator.bftWeight <= 0) {
+				throw new Error('Invalid BFT weight. BFT weight must be a positive integer.');
+			}
+			aggregateBFTWeight += validator.bftWeight;
+		}
+		if (
+			aggregateBFTWeight / BigInt(3) + BigInt(1) > precommitThreshold ||
+			precommitThreshold > aggregateBFTWeight
+		) {
+			throw new Error('Invalid precommitThreshold input.');
+		}
+		if (
+			aggregateBFTWeight / BigInt(3) + BigInt(1) > certificateThreshold ||
+			certificateThreshold > aggregateBFTWeight
+		) {
+			throw new Error('Invalid certificateThreshold input.');
+		}
+		const validatorsHash = await this._computeValidatorsHash(
+			context,
+			validators,
+			certificateThreshold,
+		);
+
+		sortValidatorsByAddress(validators);
+
+		const bftParams: BFTParameters = {
+			prevoteThreshold: (BigInt(2) * aggregateBFTWeight) / BigInt(3) + BigInt(1),
+			precommitThreshold,
+			certificateThreshold,
+			validators,
+			validatorsHash,
+		};
+
+		const votesStore = context.getStore(this.moduleID, STORE_PREFIX_BFT_VOTES);
+		const paramsStore = context.getStore(this.moduleID, STORE_PREFIX_BFT_PARAMETERS);
+		const bftVotes = await votesStore.getWithSchema<BFTVotes>(EMPTY_KEY, bftVotesSchema);
+		const nextHeight =
+			bftVotes.blockBFTInfos.length > 0
+				? bftVotes.blockBFTInfos[0].height + 1
+				: bftVotes.maxHeightPrevoted + 1;
+
+		const nextHeightBytes = intToBuffer(nextHeight, 4, BIG_ENDIAN);
+		await paramsStore.setWithSchema(nextHeightBytes, bftParams, bftParametersSchema);
+
+		const nextActiveValidators: BFTVotesActiveValidatorsVoteInfo[] = [];
+		for (const validator of validators) {
+			const existingValidator = bftVotes.activeValidatorsVoteInfo.find(v =>
+				v.address.equals(validator.address),
+			);
+			if (existingValidator) {
+				nextActiveValidators.push(existingValidator);
+				continue;
+			}
+			nextActiveValidators.push({
+				address: validator.address,
+				minActiveHeight: nextHeight,
+				largestHeightPrecommit: nextHeight - 1,
+			});
+		}
+		sortValidatorsByAddress(nextActiveValidators);
+		bftVotes.activeValidatorsVoteInfo = nextActiveValidators;
+		await votesStore.setWithSchema(EMPTY_KEY, bftVotes, bftVotesSchema);
+	}
+
+	public async getValidators(context: ImmutableAPIContext): Promise<Validator[]> {
```

Comment:
Currently, this API is not contained in the LIP. Where is it needed? Couldn't `getBFTParameters` be used instead?

Reply:
it is currently used in the fast sync logic where it needs current validator.
getBFTParameters with last block height should be give the latest one. Ill create another issue to refactor since it will affect the test a lot

Reply:
If you think it is useful, we can add the API to the LIP. Or you prefer to refactor?

Reply:
I think it would be good to add to the API in the LIP as well as current validators could be useful

Reply:
Okay, I will add the API then. Should we call it maybe `getCurrentValidators` to emphasize it is the current validator information? 

Note that when this API is called for the last block of the round at height `h` and the hook `afterBlockExecute` of the BFT module is already called, then `getCurrentValidators` will return the validators active at round `h+1` (next round). But I think this is fine and stills makes sense.

Reply:
`getCurrentValidators` sounds good, ill update everywhere!
the return value depends on the APIContext used as well. during the block execution, it will use the same context but in synchronisation, it will use different context so it will always look at the information saved to the DB